patch/attiny84.yaml

```diff
@@ -0,0 +1,221 @@
+        ADC3_ADC2_1X: [48, "Differential Inputs Positive ADC3 Negative ADC2 1x Gain"]
+        ADC3_ADC2_20X: [49, "Differential Inputs Positive ADC3 Negative ADC2 20x Gain"]
+        ADC3_ADC3_1X: [36, "Differential Inputs Positive ADC3 Negative ADC3 1x Gain"]
+        ADC3_ADC3_20X: [37, "Differential Inputs Positive ADC3 Negative ADC3 20x Gain"]
+        ADC3_ADC4_1X: [18, "Differential Inputs Positive ADC4 Negative ADC0 1x Gain"]
+        ADC3_ADC4_20X: [19, "Differential Inputs Positive ADC4 Negative ADC0 20x Gain"]
+        ADC3_ADC5_1X: [20, "Differential Inputs Positive ADC5 Negative ADC1 1x Gain"]
+        ADC3_ADC5_20X: [21, "Differential Inputs Positive ADC5 Negative ADC1 20x Gain"]
+        ADC3_ADC6_1X: [22, "Differential Inputs Positive ADC6 Negative ADC2 1x Gain"]
+        ADC3_ADC6_20X: [23, "Differential Inputs Positive ADC6 Negative ADC2 20x Gain"]
+        ADC3_ADC7_1X: [24, "Differential Inputs Positive ADC7 Negative ADC3 1x Gain"]
+        ADC3_ADC7_20X: [25, "Differential Inputs Positive ADC7 Negative ADC3 20x Gain"]
+
+        ADC4_ADC3_1X: [50, "Differential Inputs Positive ADC4 Negative ADC3 1x Gain"]
+        ADC4_ADC3_20X: [51, "Differential Inputs Positive ADC4 Negative ADC3 20x Gain"]
+        ADC4_ADC5_1X: [26, "Differential Inputs Positive ADC4 Negative ADC5 1x Gain"]
+        ADC4_ADC5_20X: [27, "Differential Inputs Positive ADC4 Negative ADC5 20x Gain"]
+
+        ADC5_ADC3_1X: [52, "Differential Inputs Positive ADC5 Negative ADC3 1x Gain"]
+        ADC5_ADC3_20X: [53, "Differential Inputs Positive ADC5 Negative ADC3 20x Gain"]
+        ADC5_ADC4_1X: [58, "Differential Inputs Positive ADC5 Negative ADC4 1x Gain"]
+        ADC5_ADC4_20X: [59, "Differential Inputs Positive ADC5 Negative ADC4 20x Gain"]
+        ADC5_ADC6_1X: [28, "Differential Inputs Positive ADC5 Negative ADC6 1x Gain"]
+        ADC5_ADC6_20X: [29, "Differential Inputs Positive ADC5 Negative ADC6 20x Gain"]
+
+        ADC6_ADC3_1X: [54, "Differential Inputs Positive ADC6 Negative ADC3 1x Gain"]
+        ADC6_ADC3_20X: [55, "Differential Inputs Positive ADC6 Negative ADC3 20x Gain"]
+        ADC6_ADC5_1X: [60, "Differential Inputs Positive ADC6 Negative ADC5 1x Gain"]
+        ADC6_ADC5_20X: [61, "Differential Inputs Positive ADC6 Negative ADC5 20x Gain"]
+        ADC6_ADC7_1X: [30, "Differential Inputs Positive ADC6 Negative ADC7 1x Gain"]
+        ADC6_ADC7_20X: [31, "Differential Inputs Positive ADC6 Negative ADC7 20x Gain"]
+
+        ADC7_ADC3_1X: [56, "Differential Inputs Positive ADC7 Negative ADC3 1x Gain"]
+        ADC7_ADC3_20X: [57, "Differential Inputs Positive ADC7 Negative ADC3 20x Gain"]
+        ADC7_ADC6_1X: [62, "Differential Inputs Positive ADC7 Negative ADC6 1x Gain"]
+        ADC7_ADC6_20X: [63, "Differential Inputs Positive ADC7 Negative ADC6 20x Gain"]
+        ADC7_ADC7_1X: [38, "Differential Inputs Positive ADC7 Negative ADC7 1x Gain"]
+        ADC7_ADC7_20X: [39, "Differential Inputs Positive ADC7 Negative ADC7 20x Gain"]
```

Comment:
I'm not sure if it is a good idea to intentionally encode enum values here that can't even fit into the field.  This will be non-obvious to downstream code, where one could miss needing to manually set the upper bits elsewhere.  Instead, I would suggest encoding the values in an enum over in the HAL, [like I did for `ATmega32U4`](https://github.com/Rahix/avr-hal/blob/master/chips/atmega32u4-hal/src/adc.rs#L7-L29) for example.

Reply:
On the tiny84 they did the right thing and moved `ADLAR` out of `ADMUX` so that the whole `MUX` field fits in one register.